framework/include/materials/MaterialPropertyInterface.h

```diff
@@ -308,6 +308,73 @@ class MaterialPropertyInterface
   /// resolve all optional properties
   virtual void resolveOptionalProperties();
 
+  /**
+   * Retrieve the property deduced from the name \p name for the specified \p material_data
```

Comment:
what is material data

docstring for it please

Reply:
so material_data is what helps you choose from face / neighbor / element right?

Reply:
Will add `that can be defined on element, face and neighboring face.` at the end.